api.md

```diff
@@ -919,6 +922,48 @@ The expected output:
 
 ---
 
+## interpolate() [](interpolate)
+
+The `interpolate` does linear interpolation for missing values.
+The `interpolate` function can only be used in an aggregation query with `time_bucket_gapfill`.
+
+#### Required Arguments [](interpolate-required-arguments)
+
+|Name|Description|
+|---|---|
+| `value` | The value to carry forward (anyelement) |
+| `prev` | The lookup expression for values before the gapfill time range |
+| `next` | The lookup expression for values after the gapfill time range |
+
+#### Sample Usage [](interpolate-examples)
+
+Get the temperature every 5 minutes for each device over the 2 weeks interpolating for missing readings:
+```sql
+SELECT
+  time_bucket_gapfill('5 minutes', time, now() - interval '2 week', now()) as time,
+  device_id,
+  interpolate(avg(temperature))
```

Comment:
I don't quite understand how the argument syntax fits in with the info from the "required arguments" section.  is `avg(temperature)` the value?  The second example arguments also seem to follow a different syntax, where the value and the two lookup expressions seem to all be nested within the `avg` function?  Can you clarify how the `interpolate` function works with `avg`, perhaps in the intro or the required arguments section?  

From the examples, it seems that the required arguments are intended to be within an aggregation function?  It may require a non-standard "required arguments" section to fully describe the behavior.

Reply:
Oh, also, it seems that the `prev` and `next` functions are "optional arguments" rather than required?

Reply:
Only value is required, the rest optional. I adjusted the docs accordingly.
The value can be any expression that results in a supported datatype. But since its an aggregation query you have to use an aggregate function on columns not referenced by the GROUP BY clause. Thats why you see the avg in the examples.